packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -2656,12 +2638,12 @@ export class ImportResolver {
         return true;
     }
 
-    private _isNativeModuleFileName(moduleName: string, fileName: string): boolean {
+    private _isNativeModuleFileName(moduleName: string, fileUri: Uri): boolean {
         // Strip off the final file extension and the part of the file name
         // that excludes all (multi-part) file extensions. This allows us to
         // handle file names like "foo.cpython-32m.so".
-        const fileExtension = getFileExtension(fileName, /* multiDotExtension */ false).toLowerCase();
-        const withoutExtension = stripFileExtension(fileName, /* multiDotExtension */ true);
+        const fileExtension = fileUri.extname.toLowerCase();
```

Comment:
There were other places where `extname` was checked against literal lowercase values. Maybe `extname` should do the `toLowerCase` call internally?

Reply:
Good idea. I'll change that.

Reply:
Actually  no we depend upon casing (at least some o the tests fail if I change it) I believe a '.Py' is not the same as a '.py' file.

Reply:
VS Code recognizes `CAPITAL.PY` as a python file and Pylance analyzes it, for what that's worth.